### PR TITLE
Adsys reworks

### PR DESCRIPTION
**Pull request by JakeDaBoss**
Adds the following:

* Improved staffwho verb that sorts staff into distinct categories: Admins, Mods, Mentors, Devs.
* Improved adminhelp system, now contains two distinct categories: ~~Gameplay/Job Inquiries and Rule issues. Routed to Mentor+ and Mod+ respectively.~~ Both categories currently go to Mentor+, by request of Spooky.
* New sound for incoming adminhelps.
* New sound for adminhelp replies. (AdminPM's).
* New sound for votes/polls.
* For all changes goes: There is currently no impediment into adminhelps. ~~Mentors just won't get rule issue ahelps. Mods and higher get adminhelps from both categories.~~
* Additionally, the old version of staffwho, adminhelps and adminpm are kept for keepsake, so that if issues pop up it will be easy to revert.

### CODE REVIEW

File: code/game/verbs/who.dm [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
 	msg += "<b>Total Players: [length(Lines)]</b>"
 	to_chat(src, msg)
-
```
Comment (NanakoAC) [inline code/game/verbs/who.dm:61]: Delete this old code

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -10,6 +10,85 @@
 		check_antagonists()
 		return
 
+	if(href_list["autoresponse"]) // new verb on the Ahelp.  Will tell the person their message was received, and they probably won't get a response
+		var/mob/ref_person = locate(href_list["autoresponse"])
+		if(!ref_person || !istype(ref_person) || !ref_person.client)
+			to_chat(usr, "\blue Looks like that person stopped existing!")
```
Comment (NanakoAC) [inline code/modules/admin/topic.dm]: Don't use byond color macros

Use the various SPAN_XXX macros, or the span(class, text) proc

In this case, SPAN_NOTICE would be best

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -10,6 +10,85 @@
...
+			if(alert(usr, "Are you sure you want to autoreply to this marked adminhelp?", "Confirmation", "Yes", "No") == "No")
+				return
+		else if (!ticket)
+			to_chat(usr, "<b>This ticket no longer exists.</b>")
+			return
+
+//		var/choice = input("Which autoresponse option do you want to send to the player?\n\n L - A webpage link.\n A - An answer to a common question.", "Autoresponse", "--CANCEL--") in list ("--CANCEL--", "IC Issue", "Being Handled", "Fixed", "Thanks", "Guilty", "L: Xeno Quickstart Guide", "L: Marine quickstart guide", "L: Current Map", "A: No plasma regen", "A: Devour as Xeno", "J: Job bans", "E: Event in progress", "R: Radios", "D: Joining disabled", "M: Macros")
+		var/choice = input("Which autoresponse option do you want to send to the player?\n\n L - A webpage link.\n A - An answer to a common question.", "Autoresponse", "--CANCEL--") in list ("--CANCEL--", "IC Issue", "Being Handled", "Fixed","Thanks!", "A: GitHub Issue", "A: Requesting to be an SCP", "A: Becoming an SCP", "A: RUST Engine", "A: D-Class Riot", "A: D-Class Cells", "A: Directions/Lost", "A: Character setup", "A: Website and Discord information")
```
Comment (NanakoAC) [inline code/modules/admin/topic.dm]: This text is SCP specific and doesn't blong here

File: code/modules/admin/topic.dm [hunk 4]
```diff
@@ -10,6 +10,85 @@
...
+
+
+		var/msgplayer
+		switch(choice)
+			if("IC Issue")
```
Comment (NanakoAC) [inline code/modules/admin/topic.dm]: String literals in switch statements are bad, make defines

File: code/modules/admin/topic.dm [hunk 5]
```diff
@@ -10,6 +10,85 @@
...
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. This issue has been deemed an IC (In-Character) issue, and will not be handled by staff. In case it's relevant, you may wish to ask your <a href='http://wiki.scp13.site/index.php?title=Rank'>Chain Of Command</a> about your issue if you believe <a href='http://wiki.scp13.site/index.php?title=Foundation_Law'>Foundation Law</a> has been broken.</b>"
+			if("Being Handled")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. The issue is already being dealt with.</b>"
+			if("Fixed")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. The issue is already fixed.</b>"
+			if("Thanks!")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. Have a SCP day!</b>"
+			if("A: Requesting to be an SCP")
```
Comment (NanakoAC) [inline code/modules/admin/topic.dm]: SCP specific content

File: code/modules/admin/topic.dm [hunk 6]
```diff
@@ -10,6 +10,85 @@
...
+			if("A: Website and Discord information")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. Hello, and welcome! Our website is located at: https://scp13.site, our wiki is located at: wiki.scp13.site, and our Discord is located at: https://discord.gg/kJ7UfvM . Hope to see you soon!</b>"
+/*			if("Thanks")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>! Have a CM day!</b>"
+			if("Guilty")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. You broke Marine Law.</b>"
+			if("L: Xeno Quickstart Guide")
+				msgplayer = "\blue <b>NOTICE: <font color=red>[usr.key]</font> is autoresponding with <font color='#009900'>'[choice]'</font>. Your answer can be found on the Xeno Quickstart Guide on our wiki. <a href='http://cm-ss13.com/wiki/Xeno_Quickstart_Guide'>Check it out here.</a></b>"
+			if("L: Marine quickstart guide")
```
Comment (NanakoAC) [inline code/modules/admin/topic.dm]: CM specific content, commented out but it shouldn't be here at all

Ask admins for correct categories to have here

File: code/game/verbs/who.dm (hunk 1 above)
Comment (JakeDaBoss) [reply code/game/verbs/who.dm:61]: That's not old code. That belongs to regular Who verb. Not staffwho.

My changes focused on staffwho.

File: code/modules/admin/verbs/adminhelp.dm [hunk 7]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
 	feedback_add_details("admin_verb","AH") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 	return
 
+*/
+
+
+//This is a list of words which are ignored by the parser when comparing message contents for names. MUST BE IN LOWER CASE!
+var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","alien","as")
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm]: use managed globals, check out global_vars/lists for examples, and put it in that folder somewhere

File: code/modules/admin/verbs/adminhelp.dm [hunk 8]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+
+/proc/generate_ahelp_key_words(var/mob/mob, var/msg)
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm:3]: This whole function is arcane, please add comments detailing what it's doing and its purpose for existing

File: code/modules/admin/verbs/adminhelp.dm [hunk 9]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+				break
+		if(!admin_found)
+			to_chat(src, "<span class='warning'>Error: Private-Message: Client not found. They may have lost connection, so please be patient!</span>")
+		return
+
+	ticket.msgs += new /datum/ticket_msg(src.ckey, null, original_msg)
+	update_ticket_panels()
+
+
+	//Options bar:  mob, details ( admin = 2, dev = 3, mentor = 4, character name (0 = just ckey, 1 = ckey and character name), link? (0 no don't make it a link, 1 do so),
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm:119]: Don't include large blocks of commented stuff

File: code/modules/admin/verbs/adminhelp.dm [hunk 10]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+				admin_number_afk++
+
+
+	switch(selected_type)
+		if("Gameplay/Job Inquiries")
+			if(mentorholders.len)
+				for(var/client/X in mentorholders) // Mentors get a message without buttons and no character name
+					if(X.get_preference_value(/datum/client_preference/staff/play_adminhelp_ping))
+						X << 'sound/effects/adminhelp_new.ogg'
+					X << mentor_msg
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm:166]: use to_chat(X, text) to send messages to a specific player

File: code/modules/admin/verbs/adminhelp.dm [hunk 11]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+			adminholders += X
+			if(X.is_afk())
+				admin_number_afk++
+
+
+	switch(selected_type)
+		if("Gameplay/Job Inquiries")
+			if(mentorholders.len)
+				for(var/client/X in mentorholders) // Mentors get a message without buttons and no character name
+					if(X.get_preference_value(/datum/client_preference/staff/play_adminhelp_ping))
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm:164]: it seems silly to have this check inside every part of the switch statement

File: code/modules/admin/verbs/adminhelp.dm [hunk 12]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+	else
+		adminmsg2adminirc(src, null, "[html_decode(original_msg)]")
+
+	feedback_add_details("admin_verb","AH") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+	return
+
+client/verb/bugreport()
+	set category = "Admin"
+	set name ="Submit Bug Report/Suggestions"
+	var url = "https://scp13.site/index.php?/forums/forum/8-bugs-and-suggestions/"
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm]: This url needs to point to our github issues page

File: code/modules/admin/verbs/adminhelp.dm [hunk 13]
```diff
@@ -141,3 +141,234 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+		if(alert("This will open the SCP13 Bug Report and Suggestions Forum Page in your Browser. Are you sure?",,"Yes","No")=="No")
+			return
+		src << link(url)
+
+client/verb/reportplayer()
+	set category = "Admin"
+	set name ="Report Player / Staff"
+	var url = "https://scp13.site/index.php?/forums/"
+	if(url)
+		if(alert("This will open the SCP13 Reports Forum Page in your Browser. This hasn't been made yet due to our Alpha stage. Are you sure?",,"Yes","No")=="No")
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm]: We don't have a forum at all, remove this verb

File: code/modules/admin/verbs/adminpm.dm [hunk 14]
```diff
@@ -185,3 +185,192 @@
 			continue
 		if(X.holder.rights & R_ADMIN|R_MOD)
 			to_chat(X, "<span class='pm'><span class='other'>" + create_text_tag("pm_other", "PM:", X) + " <span class='name'>[key_name(src, X, 0)]</span> to <span class='name'>[sender]</span>: <span class='message'>[msg]</span></span></span>")
+*/
+
+//allows right clicking mobs to send an admin PM to their client, forwards the selected mob's client to cmd_admin_pm
+/client/proc/cmd_admin_pm_context(mob/M as mob in SSmobs.mob_list)
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminpm.dm]: as long as you're replacing this, delete the old commented out version

File: code/modules/admin/verbs/adminpm.dm [hunk 15]
```diff
@@ -185,3 +185,192 @@
...
+
+		if(!msg)	return
+		if(!C)
+			if(holder)	to_chat(src, "<span class='warning'>Error: Private-Message: Client not found.</span>")
+			else		to_chat(src, "<span class='warning'>Error: Private-Message: Client not found. They may have lost connection, so try using an adminhelp!</span>")
+			return
+
+		msg = sanitize(msg)
+
+	var/datum/client_lite/receiver_lite = client_repository.get_lite_client(C)
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminpm.dm]: we don't use client_lite much, you may be better off using /datum/player

**Review by NanakoAC — CHANGES_REQUESTED** [review]
Lots of concerns here,

File: code/modules/admin/verbs/adminhelp.dm (hunk 13 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm]: I forgot to remove this.

File: code/modules/admin/verbs/adminhelp.dm (hunk 11 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm:164]: It's the only way it works, unfortunately. I've checked into it myself, and tested multiple times.

File: code/modules/admin/topic.dm (hunk 6 above)
Comment (JakeDaBoss) [reply code/modules/admin/topic.dm]: This does not matter. Autoresponses are gone.

File: code/modules/admin/topic.dm (hunk 5 above)
Comment (JakeDaBoss) [reply code/modules/admin/topic.dm]: Autoresponses remain commented out. I'll comment out the remainder.

File: code/modules/admin/verbs/adminpm.dm (hunk 15 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminpm.dm]: The old adminPM system used it too. This is why it's in there too.

File: code/modules/admin/verbs/adminpm.dm (hunk 14 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminpm.dm]: You mean the old adminpm system?

File: code/modules/admin/verbs/adminhelp.dm (hunk 10 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm:166]: I'm not sure what you mean? It works this way.

**Comment by JakeDaBoss** [comment]
Most things fixed.

File: code/modules/admin/verbs/adminhelp.dm (hunk 7 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm]: Fixed.

File: code/modules/admin/verbs/adminhelp.dm (hunk 8 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm:3]: Fixed.

File: code/modules/admin/verbs/adminhelp.dm (hunk 9 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm:119]: It's an explanation for it though? Kinda necessary for future edits.

File: code/modules/admin/verbs/adminhelp.dm (hunk 10 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminhelp.dm:166]: Tried, and did not work.

File: code/modules/admin/verbs/adminpm.dm (hunk 14 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminpm.dm]: I'd prefer to keep them, until the system's been battletested, so it's easy to revert.

Comment (NanakoAC) [reply code/modules/admin/verbs/adminpm.dm]: reverting won't be done by manually editing code, that's what we have git for

File: code/modules/admin/verbs/adminhelp.dm (hunk 10 above)
Comment (NanakoAC) [reply code/modules/admin/verbs/adminhelp.dm:166]: to_chat(X.mob, mentor_msg)

File: code/_global_vars/lists/adminhelp.dm [hunk 16]
```diff
@@ -0,0 +1,2 @@
+//This is a list of words which are ignored by the parser when comparing message contents for names. MUST BE IN LOWER CASE!
+var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","alien","as")
```
Comment (NanakoAC) [inline code/_global_vars/lists/adminhelp.dm]: This isn't a managed global, you create them with defines, like

GLOBAL_LIST_INIT(adminhelp_ignored_words, list("unknown","the","a","an","of","monkey","alien","as"))

File: code/game/verbs/who.dm (hunk 1 above)
Comment (NanakoAC) [reply code/game/verbs/who.dm:61]: the code right here says staffwho. line 62 onwards

File: code/modules/admin/topic.dm (hunk 2 above)
Comment (NanakoAC) [reply code/modules/admin/topic.dm]: this is a big block of commented out stuff, please delete it

File: code/modules/admin/verbs/adminhelp.dm [hunk 17]
```diff
@@ -1,4 +1,4 @@
-
+/*
 //This is a list of words which are ignored by the parser when comparing message contents for names. MUST BE IN LOWER CASE!
 var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","alien","as")
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm]: this list was already removed, delete this comment block

File: code/modules/admin/verbs/adminhelp.dm (hunk 9 above)
Comment (NanakoAC) [reply code/modules/admin/verbs/adminhelp.dm:119]: underneath theres a big commented block

File: code/modules/admin/verbs/adminhelp.dm [hunk 18]
```diff
@@ -141,3 +141,224 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
...
+			if(adminholders.len)
+				for(var/client/X in adminholders) // Mods
+					if(X.get_preference_value(/datum/client_preference/staff/play_adminhelp_ping))
+						X << 'sound/effects/adminhelp_new.ogg'
+					X << msg
+
+
+
+
+/*	for(var/client/X in GLOB.admins)
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminhelp.dm]: commented block

File: code/modules/admin/verbs/adminpm.dm [hunk 19]
```diff
@@ -185,3 +185,192 @@
...
+		to_chat(src, "<span class='warning'>Error: Private-Message: You are unable to use PM-s (muted).</span>")
+		return
+
+	var/msg = input(src,"Message:", "Reply private message to [sender] on IRC / 400 character limit") as text|null
+
+	if(!msg)
+		return
+
+	// Handled on Bot32's end, unsure about other bots
+//	if(length(msg) > 400) // TODO: if message length is over 400, divide it up into seperate messages, the message length restriction is based on IRC limitations.  Probably easier to do this on the bots ends.
```
Comment (NanakoAC) [inline code/modules/admin/verbs/adminpm.dm]: commented code

**Review by NanakoAC — CHANGES_REQUESTED** [review]
Lots of commented out stuff to clean up, some minor functional changes

File: code/_global_vars/lists/adminhelp.dm (hunk 16 above)
Comment (JakeDaBoss) [reply code/_global_vars/lists/adminhelp.dm]: Will go up in the next commit.

File: code/modules/admin/verbs/adminpm.dm (hunk 19 above)
Comment (JakeDaBoss) [reply code/modules/admin/verbs/adminpm.dm]: Removed. Goes up on the next commit.

**Review by NanakoAC — APPROVED** [review]

**Pull request merged**